draco/analysis/flagging.py

```diff
@@ -2981,3 +2981,259 @@ def destripe(x, w, axis=1):
+                    frac_rfi_adj[:, :, 1:], frac_rfi_adj[:, :, :-1]
+                )
+
+            frac_rfi_adj = np.transpose(frac_rfi_adj, (0, 2, 1))
+            out.add_dataset("frac_rfi")
+            out.frac_rfi[:] = frac_rfi_adj
+
+        # Return output container
+        return out
+
+
+class ReduceMaskEl(task.SingleTask):
+    """Reduce the 'el' axis from input classes and produce corresponding reduced output classes.
+
+    Reduction algorithm: If the number of True values in the mask along the el axis
+    is higher than a given threshold, set the mask to True.
+
+    Attributes
+    ----------
+    threshold : int
+    This number determines the minimum number of detected RFI events along the el axis required for a data point
+    to be included in the reduced mask. Default is 1.
+
+    """
+
+    threshold = config.Property(proptype=int, default=1)
+
+    def process(self, rfimask):
+        """Produce a RFI mask.
+
+        Parameters
+        ----------
+        rfimask : containers.LocalizedRFIMask(freq, el, time) or containers.SiderealLocalizedRFIMask(freq, ra, el)
+            El-specific RFI mask indicating channels that are free from RFI events.
+
+        Returns
+        -------
+            out : containers.RFIMask(freq, time) or containers.SiderealRFIMask(freq, ra)
+            Non el-specific RFI mask indicating channels that are free from RFI events.
+
+        """
+        # Validate inpput class
+        if (not isinstance(rfimask, containers.LocalizedRFIMask)) & (
+            not isinstance(rfimask, containers.LocalizedSiderealRFIMask)
+        ):
+            raise ValueError(
+                f"Input class must be LocalizedRFIMask or LocalizedSiderealRFIMask. Got {type(rfimask)}."
+            )
+
+        # Extract mask/frac and axes data
+        mask = rfimask.mask[:]
+        el_axis = list(rfimask.mask.attrs["axis"]).index("el")
+        freq = rfimask.freq[:]
+
+        # Apply reduction condition
+        reduced_mask = np.sum(mask, axis=el_axis) >= self.threshold
+
+        # Determine output class type
+        if isinstance(rfimask, containers.LocalizedRFIMask):
+            # LocalizedRFIMask(freq, el, time) -> RFIMask(freq, time)
+            time = rfimask.time[:]
+            output = containers.RFIMask(freq=freq, time=time)
+        elif isinstance(rfimask, containers.LocalizedSiderealRFIMask):
+            # LocalizedSiderealRFIMask(freq, ra, el)  -> SiderealRFIMask(freq, ra)
+            ra = rfimask.ra[:]
+            output = containers.SiderealRFIMask(freq=freq, ra=ra)
+
+        # The output RFI mask is not frequency distributed
+        arr = reduced_mask
+        arrdist = mpiarray.MPIArray.wrap(arr, axis=0)
+        final_mask = arrdist.allgather()
+
+        output.mask[:] = final_mask
+
+        # Return output container
+        return output
+
+
+class ApplyLocalizedRFIMask(task.SingleTask):
+    """Apply a localised (el-sensitive) RFI mask to the data by zeroing the weights.
+
```

Comment:
This docstring should mention that the data stream and mask datasets must have overlapping RA/El samples, but those samples are not assumed to be the same.